### ISSUE TITLE
CTS测试集中deeplabv3训练失败

### ISSUE THREAD

**Issue by kolinwei** [PaddlePaddle/Paddle#15677]
Python Callstacks:
  File "/root/.jumbo/lib/python2.7/site-packages/paddle/fluid/framework.py", line 1316, in append_op
    attrs=kwargs.get("attrs", None))
  File "/root/.jumbo/lib/python2.7/site-packages/paddle/fluid/layers/math_op_patch.py", line 144, in __impl__
    attrs={'axis': axis})
  File "/home/paddle/weike/cts_test/test_deeplabv3.py", line 66, in loss
    loss = loss * label_nignore
  File "/home/paddle/weike/cts_test/test_deeplabv3.py", line 85, in executor_network
    loss, mask = loss(logit, label)
  File "/home/paddle/weike/cts_test/test_deeplabv3.py", line 174, in test_train_deeplab_single_gpu
    train_info = executor_network()
  File "/root/.jumbo/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/root/.jumbo/lib/python2.7/unittest/case.py", line 327, in run
    testMethod()
  File "/root/.jumbo/lib/python2.7/unittest/case.py", line 391, in __call__
    return self.run(*args, **kwds)
  File "/root/.jumbo/lib/python2.7/site-packages/nose/case.py", line 151, in runTest
    test(result)
  File "/root/.jumbo/lib/python2.7/site-packages/nose/case.py", line 133, in run
    self.runTest(result)
  File "/root/.jumbo/lib/python2.7/site-packages/nose/case.py", line 45, in __call__
    return self.run(*arg, **kwarg)
  File "/root/.jumbo/lib/python2.7/site-packages/nose/suite.py", line 224, in run
    test(orig)
  File "/root/.jumbo/lib/python2.7/site-packages/nose/suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "/root/.jumbo/lib/python2.7/site-packages/nose/suite.py", line 224, in run
    test(orig)
  File "/root/.jumbo/lib/python2.7/site-packages/nose/suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "/root/.jumbo/lib/python2.7/site-packages/nose/suite.py", line 224, in run
    test(orig)
  File "/root/.jumbo/lib/python2.7/site-packages/nose/suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "/root/.jumbo/lib/python2.7/site-packages/nose/core.py", line 62, in run
    test(result)
  File "/root/.jumbo/lib/python2.7/site-packages/nose/core.py", line 207, in runTests
    result = self.testRunner.run(self.test)
  File "/root/.jumbo/lib/python2.7/unittest/main.py", line 95, in __init__
    self.runTests()
  File "/root/.jumbo/lib/python2.7/site-packages/nose/core.py", line 121, in __init__
    **extra_args)
  File "/root/.jumbo/bin/nosetests", line 11, in <module>
    sys.exit(run_exit())
C++ Callstacks:
Enforce failed. Expected x_dim.size() >= y_dim.size(), but received x_dim.size():2 < y_dim.size():3.
Rank of first input must >= rank of second input. at [/paddle/paddle/fluid/operators/elementwise/elementwise_op.h:54]
PaddlePaddle Call Stacks:
0       0x7ff4f5762aadp void paddle::platform::EnforceNotMet::Init<std::string>(std::string, char const*, int) + 365
1       0x7ff4f5762df7p paddle::platform::EnforceNotMet::EnforceNotMet(std::string const&, char const*, int) + 87
2       0x7ff4f6b75fa4p paddle::operators::ElementwiseOp::InferShape(paddle::framework::InferShapeContext*) const + 2836
3       0x7ff4f7278f6bp paddle::framework::OperatorWithKernel::RunImpl(paddle::framework::Scope const&, boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&) const + 603
4       0x7ff4f7276815p paddle::framework::OperatorBase::Run(paddle::framework::Scope const&, boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&) + 341
5       0x7ff4f70cd87ap paddle::framework::details::ComputationOpHandle::RunImpl() + 250
6       0x7ff4f70eb236p paddle::framework::details::OpHandleBase::Run(bool) + 118
7       0x7ff4f7077cfdp
8       0x7ff4f63874c3p std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> (), std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, void> >::_M_invoke(std::_Any_data const&) + 35
9       0x7ff4f634a7e7p std::__future_base::_State_base::_M_do_set(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>&, bool&) + 39
10        0x38c040cbe0p pthread_once + 80

**Comment by kolinwei** [comment]
经过debug是由PR14975引起的。

**Comment by kolinwei** [comment]
deeplabv3 使用fluid.ParallelExecutor时会出此问题，fluid.Executor时没问题
